leap/leap.go

```diff
@@ -0,0 +1,16 @@
+package main
+
+import "fmt"
+
+func main() {
+	fmt.Println("The program determines if a year is a leap year.")
+	fmt.Println("Enter the Year:")
+	var y int
+	fmt.Scan(&y)
+	if y%100 == 0 && y%40 == 0 {
```

Comment:
The condition is not correct. Check `1800` as an input. Your program outputs that this is a leap year, but it is false.